declarations/http.ts

```diff
@@ -1,5 +1,47 @@
 /** @noSelfInFile */
 declare namespace http {
+    type AnyCase<T extends string> = string extends T ? string : T extends `${infer F1}${infer F2}${infer R}` ? (
+        `${Uppercase<F1> | Lowercase<F1>}${Uppercase<F2> | Lowercase<F2>}${AnyCase<R>}`
+    ) :
+    T extends `${infer F}${infer R}` ? `${Uppercase<F> | Lowercase<F>}${AnyCase<R>}` :
+    "";
+
+    type HttpMethod = 'options' | 'get' | 'head' | 'post' | 'put' | 'patch' | 'delete' | 'trace' ;
+
+    /**
+     * Headers to send along with the request
+     */
+    type Headers = { [key: string]: string };
+
+    /**
+     * This table form is an expanded version of the previous syntax. All arguments from above are passed in as fields instead (for instance, http.request("https://example.com") becomes http.request { url = "https://example.com" }).
+     */
+    type HttpRequest = {
+        /**
+         * The url to request
+         */
+        url: string;
+        /**
+         * An optional string containing the body of the request. If specified, a POST request will be made instead.
+         */
+        body?: string;
+        /**
+         * Additional headers to send as part of this request.
+         */
+        headers?: Headers;
+        /**
+         * Whether to make a binary HTTP request. If true, the body will not be UTF-8 encoded, and the received response will not be decoded.
+         */
+        binary?: boolean;
+        /**
+         * Which HTTP method to use, for instance "PATCH" or "DELETE".
+         */
+        method?: AnyCase<HttpMethod>;
```

Comment:
Quick thought on this while I was reviewing it to publish to NPM, Is it truly case insensitive @unlimitedcoder2 ?

Reply:
The methods are uppercased before they are converted to the java HttpMethod [code](https://github.com/SquidDev-CC/CC-Tweaked/blob/mc-1.15.x/src/main/java/dan200/computercraft/core/apis/HTTPAPI.java#L117)

Reply:
Awsome! Nice catch!